### ISSUE TITLE
App startup time missing when using version 1.19.2

### ISSUE THREAD

**Issue by lukas-mercari** [DataDog/dd-sdk-android#1523]
**Describe what happened**
We upgraded to SDK version 1.19.2 and I noticed that the startup time metric is now gone for any app versions that use this SDK version. When I downgrade the SDK locally to 1.19.1, I can get it back.

**Steps to reproduce the issue:**
Use SDK version 1.19.2 and notice that the app startup time is empty on the DD dashboard.

<img width="279" alt="Screenshot 2023-07-05 at 13 04 13" src="https://github.com/DataDog/dd-sdk-android/assets/98808750/024c6963-de08-4265-b801-c84fce7caa07">

**Describe what you expected:**
I expect the startup time to be tracked.

**Additional context**
 - Android OS version:
 - Device Model:
 - Datadog SDK version: 1.19.2
 - Versions of any other relevant dependencies (OkHttp, …): 
 - Proguard configuration: 
 - Gradle Plugins:

**Comment by xgouchet** [comment]
Hi @lucas-mercari, we're not able to reproduce this on our side. Can you share the full configuration you're initializing the SDK with please?

**Comment by lukas-mercari** [comment]
@xgouchet Thank you for taking a look! This is how we initialise the SDK:
```
        val configuration = Configuration.Builder(
            rumEnabled = true,
            logsEnabled = true,
            tracesEnabled = true,
            crashReportsEnabled = true
        )
            .trackInteractions()
            .trackLongTasks(longTaskThresholdMs = 100L)
            .setWebViewTrackingHosts(...)
            .useViewTrackingStrategy(
                ActivityViewTrackingStrategy(
                    trackExtras = true,
                    componentPredicate = object : ComponentPredicate<Activity> {
                        override fun accept(component: Activity) =
                            !Class.forName("OurMainActivity").kotlin.isInstance(component)

                        override fun getViewName(component: Activity): String? = null
                    }
                )
            )
            .useSite(DatadogSite.US1)
            .sampleRumSessions(...)
            .build()
        val credentials = Credentials(
            clientToken = clientToken,
            envName = ENVIRONMENT,
            variant = BuildConfig.BUILD_TYPE,
            rumApplicationId = APPLICATION_ID
        )
        Datadog.initialize(context, credentials, configuration, TrackingConsent.GRANTED)
        GlobalRum.registerIfAbsent(RumMonitor.Builder().build())
```

**Comment by 0xnm** [comment]
Hello @lukas-mercari! I cannot reproduce this in our sample app using the configuration you've provided.

Can you please give us more details:

* Do you integrate Datadog SDK in the application directly or in the library?
* Do you start your application in foreground or it happens in background?
* Do you see actions with `@action.type:application_start` in your dashboard?

Also it is worth to mention, that version `1.19.1` shouldn't be used as it is mentioned [here](https://github.com/DataDog/dd-sdk-android/releases/tag/1.19.1).

**Comment by lukas-mercari** [comment]
@0xnm Sorry for the late reply.

- We integrate Datadog in an application.
- Typically foreground (although there are starts on background as well, e.g. for push notifications. Note: we don't want to enable `trackBackgroundEvents`.)
- Indeed, we don't get any `@action.type:application_start` for recent versions. (Note: Sessions often start with a  `ApplicationLaunch` screen though.)

**Comment by 0xnm** [comment]
Hello @lukas-mercari! Sorry for the late reply. I think that your issue may be the same as reported in #1582. We have a fix ready, you can check more details in the related issue and PR.

**Comment by lukas-mercari** [comment]
Thank you @0xnm ! Yes, it's the same issue :) Let me close this report!

**Comment by 0xnm** [comment]
Fix is now released in version [2.1.0](https://github.com/DataDog/dd-sdk-android/releases/tag/2.1.0), please check it out and let us know if you have any issues / anomalies.
